src/slick.grid.ts

```diff
@@ -3837,27 +3860,29 @@ export class SlickGrid<TData = any, C extends Column<TData> = Column<TData>, O e
         }
 
         if (this.hasFrozenColumns() && (i > this._options.frozenColumn!)) {
-          this.appendCellHtml(stringArrayR, row, i, (colspan as number), d);
+          //could add it as a check in the if the !
+          this.appendCellHtml(tmpRowR!, row, i, (colspan as number), d);
         } else {
-          this.appendCellHtml(stringArrayL, row, i, (colspan as number), d);
+          this.appendCellHtml(tmpRow, row, i, (colspan as number), d);
         }
       } else if (m.alwaysRenderColumn || (this.hasFrozenColumns() && i <= this._options.frozenColumn!)) {
-        this.appendCellHtml(stringArrayL, row, i, (colspan as number), d);
+        this.appendCellHtml(tmpRow, row, i, (colspan as number), d);
       }
 
       if ((colspan as number) > 1) {
         i += ((colspan as number) - 1);
       }
     }
-
-    stringArrayL.push('</div>');
+    if(!this._options.nonce){
+      (stringArrayL as string[]).push('</div>');
 
     if (this.hasFrozenColumns()) {
-      stringArrayR.push('</div>');
+      (stringArrayR as string[]).push('</div>');
+    }
```

Comment:
there seems to be an indentation issue somewhere, this curly brace and the next line are aligned and they shouldn't be, please fix indentation